build.gradle.kts

```diff
@@ -1,38 +1,61 @@
 plugins {
-	id("maven-publish")
+	id("io.github.gradle-nexus.publish-plugin") version "1.0.0"
 	id("io.gitlab.arturbosch.detekt").version(Dependencies.detektVersion)
 }
 
 // Versioning
 allprojects {
 	group = "org.jellyfin.apiclient"
-	version = getProperty("jellyfin.version")?.removePrefix("v") ?: "SNAPSHOT"
+	version = getProperty("jellyfin.version")?.removePrefix("v") ?: "latest-SNAPSHOT"
 }
 
 buildscript {
-	repositories {
-		google()
-		jcenter()
-	}
+	repositories.defaultRepositories()
 
 	dependencies {
 		classpath(Dependencies.Android.buildTools)
 		classpath(Dependencies.Kotlin.gradlePlugin)
 	}
 }
 
-allprojects {
-	repositories {
-		google()
-		jcenter()
-	}
+// Add Sonatype publishing repository
+nexusPublishing.repositories.sonatype {
+	nexusUrl.set(uri("https://s01.oss.sonatype.org/service/local/"))
+	snapshotRepositoryUrl.set(uri("https://s01.oss.sonatype.org/content/repositories/snapshots/"))
+
+	username.set(project.properties["ossrh.username"].toString())
+	password.set(project.properties["ossrh.password"].toString())
+}
 
-	// Publishing
-	plugins.apply("maven-publish")
-	publishing.repositories.jellyfinBintray(this)
+subprojects {
+	// Enable required plugins
+	apply<MavenPublishPlugin>()
+	apply<SigningPlugin>()
+	apply<io.gitlab.arturbosch.detekt.DetektPlugin>()
+
+	// Add dependency repositories
+	repositories.defaultRepositories()
+
+	// Run block after creating project specific configuration
+	afterEvaluate {
+		// Add signing config
+		configure<SigningExtension> {
+			val signingKey = project.properties["signing.key"]?.toString()
+			val signingPassword = project.properties["signing.password"]?.toString()
+
+			useInMemoryPgpKeys(signingKey, signingPassword)
+			val publishing: PublishingExtension by project
+			sign(publishing.publications)
+		}
+
+		// Add POM to projects that use publishing
+		configure<PublishingExtension> {
```

Comment:
You should be able to use a `publishing` block instead, reads a bit nicer. Same for `signing`. [This](https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:deferred_configuration) might also help.

Reply:
Using `publishing {}` directly actually caused issues in other files, I still don't exactly know why. I've started to like the `configure<>` syntax though :p